crates/bevy_ecs/src/reflect/resource.rs

```diff
@@ -65,15 +67,15 @@ impl ReflectResourceFns {
     ///
     /// This is useful if you want to start with the default implementation before overriding some
     /// of the functions to create a custom implementation.
-    pub fn new<T: Resource + Reflect + FromWorld>() -> Self {
+    pub fn new<T: Resource + Reflect + FromReflect>() -> Self {
```

Comment:
Nit: We can remove the `Reflect` bound since `FromReflect: Reflect`
```suggestion
    pub fn new<T: Resource + FromReflect>() -> Self {
```